### PR TITLE
Split authentication and authorization into two separate docs

### PR DESCRIPTION
**Pull request by shimi9276**
As first step to adding documentation on user management.

### CODE REVIEW

File: docs/reference/authentication.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+### User Authentication
+
+lakeFS authenticates users from a built-in authentication database, or
+optionally from a configured LDAP server.
+
+#### Built-in database
+
+The built-in authentication database is always present and active.  Use the
+Web UI at Administration / Users to create users.  Users have an access key
+`AKIA...` and an associated secret access key.  These credentials are valid
```
Comment (johnnyaug) [inline docs/reference/authentication.md:27]: Don't we have email authentication as well now?

File: docs/reference/authentication.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+to log into the Web UI, or to authenticate programmatic requests to the API
+Server, or for S3 Gateway Authentication.
```
Comment (johnnyaug) [inline docs/reference/authentication.md:29]:
```suggestion
to log into the Web UI, or to authenticate programmatic requests to the API
Server or the S3 Gateway.
```

**Review by johnnyaug — COMMENTED** [review]
Can we add info about the user/password authentication? Assuming it's available today - I'm not really in the loop there.

File: docs/reference/authentication.md (hunk 1 above)
Comment (shimi9276) [reply docs/reference/authentication.md:27]: We do, The idea here is to add exactly that (in a coming PR hopefully by today/tomorrow), and as a first step we felt is made more sense to split the authentication and authorization into two separate docs.

**Pull request merged**